HPCC-33041 ECL Watch UI test (GH Action) reports an error with 'Files' page.

AttilaVamos opened the pull request:
ActivitiesTest.java:
- Change the 'should match' behaviour to a 'looks like' attempt when the test tries to identify a WEB page where it is landed after clicked a hyperlink..

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->
Tested locally.
<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

cloLN reviewed (approved):
Changes look reasonable.

AttilaVamos commented:
@ghalliday please merge.

ghalliday commented:
@AttilaVamos please rebase to avoid the merge conflicts.  Thanks.

ghalliday commented on .github/workflows/test-ui-gh_runner.yml:
```diff
@@ -148,7 +148,8 @@ jobs:
           wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
           sudo apt-get install -y ./google-chrome-stable_current_amd64.deb
-          wget https://storage.googleapis.com/chrome-for-testing-public/126.0.6478.126/linux64/chromedriver-linux64.zip
+          #wget https://storage.googleapis.com/chrome-for-testing-public/126.0.6478.126/linux64/chromedriver-linux64.zip
```
should this be removed?

AttilaVamos replied:
Yes, removed

ghalliday commented on .github/workflows/test-ui-gh_runner.yml:
```diff
@@ -185,7 +186,7 @@ jobs:
         uses: actions/upload-artifact@v4
         with:
-          name: ${{ inputs.asset-name }}-ecl_watch_ui_tests
+          name: ${{ inputs.asset-name }}-ecl_watch_ui_tests_logs
```
is this change deliberate?

ghalliday commented on esp/src/test-ui/tests/framework/pages/ActivitiesTest.java:
```diff
@@ -39,19 +39,28 @@ public void testActivitiesPage() {
+        
         for (NavigationWebElement element : navWebElements) {
 
+            StringBuilder erorMmsg = new StringBuilder("OK");
```
trivial: typo errorMsg

AttilaVamos replied:
Fixed.

ghalliday commented on esp/src/test-ui/tests/framework/pages/ActivitiesTest.java:
```diff
@@ -80,19 +89,29 @@ private String getCurrentPage() {
+        boolean retVal = true;
+                
         List<String> tabsList = URLConfig.tabsListMap.get(element.name());
+        int numOfElemets = tabsList.size();
```
typo: numOfEleme**n**ts

AttilaVamos replied:
Fixed

ghalliday commented on esp/src/test-ui/tests/framework/pages/ActivitiesTest.java:
```diff
@@ -80,19 +89,29 @@ private String getCurrentPage() {
-
-        return true;
+        
+        if ( (elementFound / numOfElemets) < 0.5 ) {
```
Need to protect against numElements == 0 (and return early?)

AttilaVamos replied:
Changed

ghalliday reviewed (commented):
@AttilaVamos a couple of comments/questions

ghalliday reviewed (approved):
Looks good to me.  Please squash when ready

cloLN reviewed (approved):
Looks good to me.

AttilaVamos committed:
HPCC-33041 ECL Watch UI test (GH Action) reports an error with 'Files' page.

ActivitiesTest.java:
- Change the 'should match' behaviour to a 'looks like' attempt when the test
  tries to identify a WEB page where it is landed after clicked a hyperlink..

test-ui-gh_runner.yml:
- Change Chromedriver vesion (PR-19330 not merged yet)
- Fix the tests log artifact name

Tested locally.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

AttilaVamos commented:
@ghalliday squashed and please merge.
